db-tools-ij/src/main/java/com/splicemachine/db/impl/tools/ij/ijImpl.java

```diff
@@ -31,6 +36,30 @@
     String protocol = null; // the (single) unnamed protocol
     Hashtable namedProtocols;
 
+    // used in utilMain to cancel currently running statements with Ctrl+C
+    // currentStatement can be accessed by the processing thread and the signal handler thread.
+    // this is not a high concurrency situation, still we make most accesses synchronized.
+    Statement currentStatement = null;
+    Object currentStatementLock = new Object();
+
+    public void cancelCurrentStatementOrExit(LocalizedOutput out,
+                                                SignalHandler originalSigIntHandler,
+                                                Signal sigInt) {
+        synchronized (currentStatementLock) {
+            try {
+                if(currentStatement == null) {
+                    out.println("\nCTRL+C -> exit...\n");
+                    originalSigIntHandler.handle(sigInt);
+                }
+                else {
+                    out.println("\nCTRL+C -> Canceling Statement...\n");
+                    currentStatement.cancel();
+                }
+            } catch (SQLException e) {
+                out.println("Error Canceling Statement: " + e.toString());
```

Comment:
i just looked this up, seems like Canceling is American English and Cancelling is British English, see e.g. https://en.wiktionary.org/wiki/canceling#English